src/middlewared/middlewared/plugins/keychain.py

```diff
@@ -23,13 +24,84 @@ class KeychainCredentialType:
 
     credentials_schema = NotImplemented
 
+    used_by_delegates = []
+
     async def validate_and_pre_save(self, middleware, verrors, schema_name, attributes):
         pass
 
-    async def used_by(self, middleware, id):
+
+class KeychainCredentialUsedByDelegate:
+    unbind_method = NotImplemented
+
+    def __init__(self, middleware):
+        self.middleware = middleware
+
+    async def query(self, id):
+        raise NotImplementedError
+
+    async def get_title(self, row):
+        raise NotImplementedError
+
+    async def unbind(self, row):
+        raise NotImplementedError
+
+
+class KeychainCredentialUsedByDelegateUnbindMethod(enum.Enum):
+    DELETE = "delete"
+    DISABLE = "disable"
+
+
+class OtherKeychainCredentialKeychainCredentialUsedByDelegate(KeychainCredentialUsedByDelegate):
```

Comment:
KeychainCredentialKeychainCredential ?

Reply:
KeychainCredentialUsedByDelegate for keychain credential that is used by Other Keychain Credential